osier/technology.py

```diff
@@ -327,17 +435,43 @@ def annual_fixed_cost(self):
 
     @property
     def variable_cost(self):
-        return self.fuel_cost + self.om_cost_variable
+        """
+        Combines the fuel and variable operating costs into a total variable cost
+        associated with technology usage. 
+        
+        Notes
+        -----
+        This function will attempt to merge the two values, even if they have 
+        different sizes and types. Therefore it is recommended that users
+        pass values of the same size and type to prevent unexpected behavior.
+        """
+        if (isinstance(self.fuel_cost, _constant_types) and isinstance(self.om_cost_variable, _constant_types)):
+            return self.fuel_cost + self.om_cost_variable
+        elif (isinstance(self.fuel_cost, _array_types) and isinstance(self.om_cost_variable, _constant_types)):
+            return self.fuel_cost + np.ones(len(self.fuel_cost)) * self.om_cost_variable
+        elif (isinstance(self.fuel_cost, _constant_types) and isinstance(self.om_cost_variable, _array_types)):
+            return self.fuel_cost * np.ones(len(self.om_cost_variable)) + self.om_cost_variable
+        elif (isinstance(self.fuel_cost, _constant_types) and isinstance(self.om_cost_variable, _array_types)):
+            return self.fuel_cost * np.ones(len(self.om_cost_variable)) + self.om_cost_variable
+        elif (isinstance(self.fuel_cost, _array_types) and isinstance(self.om_cost_variable, _array_types)):
+            min_len = min(len(self.fuel_cost), len(self.om_cost_variable))
+            return self.fuel_cost[:min_len] + self.om_cost_variable[:min_len]
+        else:
+            raise TypeError(f"Fuel cost has type <{type(self.fuel_cost)}>.\n"+
+                            f"OM variable cost has type <{type(self.om_cost_variable)}>.\n"
+                            "One or both of these types are unknown.")
+        
 
     def variable_cost_ts(self, size):
         """
-        Returns the total variable cost as a time series of
+        Returns the total variable cost as an array of
         length :attr:`size`.
 
         .. warning::
-            The current implementation assumes a single constant cost
-            for the variable cost. In the future, users will be able to
-            pass their own time series data.
+            The current implementation will only select the 
```

Comment:
Think you may need a new line in between `.. warning::` and the actual text

Reply:
Nope, see [sphinx docs](https://sublime-and-sphinx-guide.readthedocs.io/en/latest/notes_warnings.html)